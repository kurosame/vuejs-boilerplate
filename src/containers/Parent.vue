<template>
  <div>
    <child :count="counter.count" @add-count="addCount"></child>
    <child :count="counter.axiosCount" @add-count="addAxiosCount"></child>
    <child
      :count="counter.asyncAwaitCount"
      @add-count="addAsyncAwaitCount"
    ></child>
  </div>
</template>
<script lang="ts">
import Vue from 'vue'
import { mapActions } from 'vuex'
import Child from '@/components/Child.vue'
import { CounterState } from '@/vuex/modules/counter'
import { ADD_ASYNC_AWAIT_COUNT, ADD_AXIOS_COUNT, ADD_COUNT } from '@/vuex/types'

export default Vue.extend({
  name: 'Parent',
  components: { Child },
  computed: {
    counter(): CounterState {
      return this.$store.getters.counter
    }
  },
  methods: {
    ...mapActions({
      addCount: ADD_COUNT,
      addAxiosCount: ADD_AXIOS_COUNT,
      addAsyncAwaitCount: ADD_ASYNC_AWAIT_COUNT
    })
  }
})
</script>
