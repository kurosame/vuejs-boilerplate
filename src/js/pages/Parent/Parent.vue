<template src="./parent.html"></template>
<style scoped lang="scss" src="./parent.scss"></style>
<script src="./parent.js"></script>
