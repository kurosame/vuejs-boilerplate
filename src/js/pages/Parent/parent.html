<div class="parent">
  <!-- vuex-router-sync sample -->
  {{ $store.state.route.name }}
  <child :addValue="addValue" :axiosSample="axiosSample" :asyncAwaitSample="asyncAwaitSample" :count="count" :axiosCount="axiosCount"
    :asyncCount="asyncCount">
  </child>
</div>