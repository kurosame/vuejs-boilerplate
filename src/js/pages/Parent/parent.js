import { mapActions, mapGetters } from 'vuex'
import { ADD_VALUE, AXIOS_SAMPLE, ASYNC_AWAIT_SAMPLE } from '@/vuex/types'
import Child from '@/components/Child/Child.vue'

export default {
  name: 'Parent',
  methods: {
    ...mapActions({
      addValue: ADD_VALUE,
      axiosSample: AXIOS_SAMPLE,
      asyncAwaitSample: ASYNC_AWAIT_SAMPLE
    })
  },
  computed: {
    ...mapGetters(['count', 'axiosCount', 'asyncCount'])
  },
  components: {
    Child
  }
}
