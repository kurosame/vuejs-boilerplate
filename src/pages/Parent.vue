<template>
  <div class="parent">
    Parent
    <child :count="count" :axiosCount="axiosCount" :asyncAwaitCount="asyncAwaitCount"
      @addCount="addCount" @addAxiosCount="addAxiosCount" @addAsyncAwaitCount="addAsyncAwaitCount">
    </child>
  </div>
</template>
<style lang="scss" scoped>
.parent {
  color: white;
  background-color: green;
}
</style>
<script lang="ts">
/* tslint:disable:prettier */
// This rule is temporary solution to bug by TSLint (tslint-loader) of vue-loader
/* tslint:enable:prettier */
import Child from '@/components/Child.vue'
import { ADD_ASYNC_AWAIT_COUNT, ADD_AXIOS_COUNT, ADD_COUNT } from '@/vuex/types'
import Vue from 'vue'
import { mapActions, mapGetters } from 'vuex'

export default Vue.extend({
  name: 'Parent',
  methods: {
    ...mapActions({
      addCount: ADD_COUNT,
      addAxiosCount: ADD_AXIOS_COUNT,
      addAsyncAwaitCount: ADD_ASYNC_AWAIT_COUNT
    })
  },
  computed: {
    ...mapGetters(['count', 'axiosCount', 'asyncAwaitCount'])
  },
  components: {
    Child
  }
})
</script>
