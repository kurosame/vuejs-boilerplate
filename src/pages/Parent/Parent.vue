<template src="./parent.html"></template>
<style scoped src="./parent.css"></style>
<script src="./parent.js"></script>
