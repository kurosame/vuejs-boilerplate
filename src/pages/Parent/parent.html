<div class="parent">
  <!-- vuex-router-sync sample -->
  {{ $store.state.route.name }}
  <child
    :addValue="addValue"
    :count="count"
  >
  </child>
</div>
