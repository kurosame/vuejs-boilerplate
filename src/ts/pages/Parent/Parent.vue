<template src="./parent.html"></template>
<style scoped src="./parent.scss"></style>
<script lang="ts" src="./parent.ts"></script>
