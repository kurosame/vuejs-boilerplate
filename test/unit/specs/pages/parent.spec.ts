import { mount } from '@vue/test-utils'
import Parent from '@/pages/Parent.vue'

describe('pages', () => {
  describe('Parent.vue', () => {
    // const wrapper = mount(Parent)

    test('count', () => {})
  })
})
