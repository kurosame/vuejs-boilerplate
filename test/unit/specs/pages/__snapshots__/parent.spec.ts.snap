// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `
<div class="parent">
  Parent
  <child-stub></child-stub>
</div>
`;
